src/tz_macos.rs

```diff
@@ -1,80 +1,135 @@
+fn get_timezone() -> Option<String> {
     // The longest name in the IANA time zone database is 25 ASCII characters long.
     const MAX_LEN: usize = 32;
+    let mut buf = [0; MAX_LEN];
 
     // Get system time zone, and borrow its name.
-    let tz = Dropping::new(CFTimeZoneCopySystem())?;
-    let name = CFTimeZoneGetName(tz.0);
-    if name.is_null() {
-        return None;
-    }
+    let tz = system_time_zone::SystemTimeZone::new()?;
+    let name = tz.name()?;
 
     // If the name is encoded in UTF-8, copy it directly.
-    let cstr = CFStringGetCStringPtr(name, kCFStringEncodingUTF8);
-    if !cstr.is_null() {
-        let cstr = std::ffi::CStr::from_ptr(cstr);
-        if let Ok(name) = cstr.to_str() {
-            return Some(name.to_owned());
-        }
-    }
-
-    // Otherwise convert the name to UTF-8.
-    let mut buf = [0; MAX_LEN];
-    let mut buf_bytes = 0;
-    let range = CFRange {
-        location: 0,
-        length: CFStringGetLength(name),
+    let name = if let Some(name) = name.as_utf8() {
+        name
+    } else {
+        // Otherwise convert the name to UTF-8.
+        name.to_utf8(&mut buf)?
     };
-    if CFStringGetBytes(
-        name,
-        range,
-        kCFStringEncodingUTF8,
-        b'\0',
-        false as Boolean,
-        buf.as_mut_ptr(),
-        buf.len() as isize,
-        &mut buf_bytes,
-    ) != range.length
-    {
-        // Could not convert the name.
-        None
-    } else if buf_bytes < 1 || buf_bytes >= MAX_LEN as isize {
+
+    if name.is_empty() || name.len() >= MAX_LEN {
         // The name should not be empty, or excessively long.
         None
     } else {
-        // Convert the name to a `String`.
-        let name = core::str::from_utf8(&buf[..buf_bytes as usize]).ok()?;
         Some(name.to_owned())
     }
 }
 
-struct Dropping<T>(*const T);
+mod system_time_zone {
+    //! create a safe wrapper around `CFTimeZoneRef`
+
+    use core_foundation_sys::base::{CFRelease, CFTypeRef};
+    use core_foundation_sys::timezone::{CFTimeZoneCopySystem, CFTimeZoneGetName, CFTimeZoneRef};
+
+    pub(crate) struct SystemTimeZone(CFTimeZoneRef);
 
-impl<T> Drop for Dropping<T> {
-    #[inline]
-    fn drop(&mut self) {
-        unsafe { CFRelease(self.0 as CFTypeRef) };
+    impl Drop for SystemTimeZone {
+        fn drop(&mut self) {
+            // SAFETY: `SystemTimeZone` is only ever created with a valid `CFTimeZoneRef`.
+            unsafe { CFRelease(self.0 as CFTypeRef) };
+        }
+    }
+
+    impl SystemTimeZone {
+        pub(crate) fn new() -> Option<Self> {
+            // SAFETY: No invariants to uphold. We'll release the pointer when we don't need it anymore.
+            let v: CFTimeZoneRef = unsafe { CFTimeZoneCopySystem() };
+            if v.is_null() {
+                None
+            } else {
+                Some(SystemTimeZone(v))
+            }
+        }
+
+        pub(crate) fn name(&self) -> Option<super::string_ref::StringRef<Self>> {
```

Comment:
Could you explicitly specify lifetime parameters here to clarify that the `StringRef` is bound to `&self`?

Reply:
I added some documentation about this in 8b9c9f0af2b35e8a083a128f539997849241d294. Is that what you had in mind?